cmake/compiler/gcc/target.cmake

```diff
@@ -5,7 +5,14 @@ set_ifndef(C++ g++)
 # Configures CMake for using GCC, this script is re-used by several
 # GCC-based toolchains
 
-find_program(CMAKE_C_COMPILER ${CROSS_COMPILE}${CC} PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+if("${SPARSE}" STREQUAL "y")
+  find_program(CMAKE_C_COMPILER cgcc)
+  find_program(REAL_CC ${CROSS_COMPILE}${CC} PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+  set(ENV{REAL_CC} ${REAL_CC})
+else()
+  find_program(CMAKE_C_COMPILER ${CROSS_COMPILE}${CC} PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+endif()
```

Comment:
Unnecessary duplication of the `find_program` for the C compiler.

Please leave the existing `find_program` call as is and only add special handling for the `SPARSE` case by setting `REAL_CC` to `${CMAKE_C_COMPILER}`. Also no need for `else()` then.

Reply:
@stephanosio are you sure? For sparse we need to set `CMAKE_C_COMPILER` to cgcc, not to the toolchain compiler

Reply:
I meant something like:
```
find_program(CMAKE_C_COMPILER ${CROSS_COMPILE}${CC} PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
if("${SPARSE}" STREQUAL "y")
  set(ENV{REAL_CC} ${(CMAKE_C_COMPILER})
  find_program(CMAKE_C_COMPILER cgcc)
endif()
```

Reply:
@stephanosio ok, sure, this can be done. I pushed an update

Reply:
@stephanosio in fact, I take it back, that doesn't work - once a variable is set `find_program()` doesn't overwrite it. That is documented in cmake docs https://cmake.org/cmake/help/latest/command/find_program.html, but they also say, that clearing that variable should help. Not sure what they mean by that but `unset()` doesn't seem to help either. Reverting to the original version.

Reply:
> Not sure what they mean by that but `unset()` doesn't seem to help either.

Fair enough. We can refactor this later. cc @tejlmand